Library/Homebrew/utils/curl.rb

```diff
@@ -19,6 +20,8 @@ def curl(*args)
 end
 
 def curl_output(*args)
-  curl_args = curl_args(args) - ["--fail"]
-  Utils.popen_read_text(*curl_args)
+  curl_args = curl_args(args) - ["--fail", "--silent"]
+  Open3.popen3(*curl_args) do |_, stdout, stderr, status|
```

Comment:
Unfortunately, I'm not aware of a somewhat elegant solution …

We could re-implement `Open3.popen3` adapting it from a more modern Ruby, as I believe all the building blocks exist, but this surely feels like a bad solution.